sdk/src/transaction_context.rs

```diff
@@ -538,6 +611,28 @@ impl<'a> BorrowedAccount<'a> {
 
     /// Assignes the owner of this account (transaction wide)
     pub fn set_owner(&mut self, pubkey: &[u8]) -> Result<(), InstructionError> {
```

Comment:
(I'm unresolving this review comment again).

i don't think this is addressed, right?

Reply:
The `set_data_length` method is a good example of what I'm proposing. It has the equality check inside.

Reply:
The check is already there, see line 633: `if self.get_owner().to_bytes() == pubkey`

Reply:
Oh ok, I missed that you added it there. I would like it to be the first check in the method for `set_lamports` and `set_owner` so that the equality check at the call site can be removed. Same applies for `set_data` but with the extra check to avoid expensive comparison if possible.

Reply:
Already discussed on Slack, but for completeness. The check at the call site is necessary and can not be moved inside the callee, and the check in the callee can not be moved to be first thing to make the check at the call site redundant. That is because ABIv1 and ABIv2 differ in how they deal with faulty accesses. In ABIv1 the runtime always tries to write to all accounts, but only throws errors if the data written is actually different as a workaround. ABIv2 does not do that anymore and always checks the permissions, even if the data written is the same as it already is.

Reply:
Sounds good, thanks for the discussion!